server/admin/tree_gc.go

```diff
@@ -100,7 +94,13 @@ func (gc *DeletedTreeGC) Run(ctx context.Context) {
 		}
 
 		d := gc.minRunInterval + time.Duration(rand.Int63n(gc.minRunInterval.Nanoseconds()))
-		timeSleep(d)
+		select {
+		case <-ctx.Done():
+			return
+		case <-timeAfter(d):
+		default:
```

Comment:
I suspect you probably don't want the `default:` case in there? I think having it will defeat the new `timeAfter()` sleep and cause the loop to "busy spin" until `ctx` is done (i.e. I think you want this `select` to block until either `ctx` is done or `d` time has passed?)

Reply:
Yes, of course.  I still find `select` a bit tricky :sweat_smile:

Reply:
Fixed

Reply:
> Yes, of course. I still find `select` a bit tricky 😅

No worries, it happens to us all sometime or other :)